pallets/order-book/src/lib.rs

```diff
@@ -553,23 +563,82 @@ impl<T: Config> Pallet<T> {
 
         Ok(())
     }
+}
+
+// todo: make pub(tests) (k.ivanov)
+pub trait CurrencyLocker<AccountId, AssetId, DEXId> {
```

Comment:
tbh I don't think trait is needed here; it doesn't contribute to loose coupling, just adds redundant code

Reply:
@wer1st what do you think?

Reply:
You can leave it here, I will move it in `traits.rs` in my branch

Reply:
Ok, my concern was about whether it is needed at all. However, now I think about it, it will be provided to order book datastructure, so it makes sense at the end